packages/create-react-app/createReactApp.js

```diff
@@ -606,7 +606,7 @@ function isSafeToCreateProjectIn(root, name) {
 }
 
 function checkIfOnline(useYarn) {
-  if (!useYarn) {
+  if (!useYarn || process.env.https_proxy) {
```

Comment:
That was my first though too. However, to test whether yarn has access vs the node process has access are quite different things and one doesn't necessarily prove the other.
Similarly, being able to resolve the address in DNS doesn't prove yarn will work either, which made me consider the purpose of this check. My conclusion was that this is an aid to make a best effort if there isn't an internet connection.
My thinking from there was that in its current state, you can't use create-react-app if you need a proxy, regardless of settings.  With this change at worst you'll get a different error.